### PR TITLE
5256.dev 2

### PR DESCRIPTION
**Pull request by VRiabovol**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,7 +19,8 @@
 web/themes/custom/mytheme/dist/javascript.js
 /web/sites/default/settings.local.php
 /web/sites/default/services.local.yml
-web/themes/custom/mytheme/dist/
+/web/themes/custom/mytheme/dist/
+/web/modules/custom/drupal_bootcamp_weather
```
Comment (boddy-devbranch) [inline .gitignore:23]: I did not get it.

File: web/modules/custom/weather/src/Form/APIForm.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('City and country name'),
+    ];
+    $form['key'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('API key'),
+    ];
```
Comment (boddy-devbranch) [inline web/modules/custom/weather/src/Form/APIForm.php]: I assumed there are saved values required. please add them

File: web/modules/custom/weather/src/Form/APIForm.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    return $form;
+  }
+
+  /**
+   * Rewrite method from interface.
+   *
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    if (strlen($form_state->getValue('name')) < 1) {
```
Comment (boddy-devbranch) [inline web/modules/custom/weather/src/Form/APIForm.php]: use empty() please

File: web/modules/custom/weather/src/Form/APIForm.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      $form_state->setErrorByName('name', $this->t('City and country should have any symbols.'));
+    }
+    elseif (strlen($form_state->getValue('key')) < 1) {
```
Comment (boddy-devbranch) [inline web/modules/custom/weather/src/Form/APIForm.php]: use empty() please

File: web/modules/custom/weather/src/Form/APIForm.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    $this->messenger()->addStatus($this->t('Your API key API key successfully added.'));
+  }
+
+  /**
+   * Validation have a string digits.
+   */
+  public function haveDigitValidation(FormStateInterface $form_state) {
+    $name = $form_state->getValue('name');
+    $result = FALSE;
+    for ($i = 0; strlen($name) >= $i; $i++) {
```
Comment (boddy-devbranch) [inline web/modules/custom/weather/src/Form/APIForm.php]:
```
if (preg_match('~[0-9]+~', $string)) {
    echo 'string with numbers';
}
```

File: web/modules/custom/weather/src/Plugin/Block/WeatherBlock.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      $data = json_decode($response->getBody()->getContents(), TRUE);
+      foreach ($data as $key => $value) {
+        if ($key == "location") {
+          $result['name'] = $data["location"]["name"];
+        }
+        elseif ($key == "current") {
+          $result['temperature'] = $data["current"]["temp_c"];
+          $result['icon'] = $data["current"]["condition"]["icon"];
+        }
+      }
```
Comment (boddy-devbranch) [inline web/modules/custom/weather/src/Plugin/Block/WeatherBlock.php]:
```
$result['name'] = $data["location"]["name"] ?? '';
$result['temperature'] = $data["current"]["temp_c"] ?? '';
$result['icon'] = $data["current"]["condition"]["icon"] ?? '';
```

**Review by boddy-devbranch — DISMISSED** [review]

**Review by boddy-devbranch — CHANGES_REQUESTED** [review]
see comments

File: web/modules/custom/weather/src/Form/APIForm.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    elseif ($this->keyNameValidation($form_state)) {
+      $form_state->setErrorByName('key_verification', $this->t('Wrong API key.'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->messenger()->addStatus($this->t('Your API key API key successfully added.'));
```
Comment (boddy-devbranch) [inline web/modules/custom/weather/src/Form/APIForm.php:91]: There is no code for saving settings.

File: .gitignore (hunk 1 above)
Comment (VRiabovol) [reply .gitignore:23]: This is my first module, I deleted it in php storm and had a problem. Then I fixed problem and decide to  create a new module, these add to ignore...

File: web/modules/custom/weather/src/Form/APIForm.php (hunk 2 above)
Comment (VRiabovol) [reply web/modules/custom/weather/src/Form/APIForm.php:49]: +

**Review by boddy-devbranch — APPROVED** [review]

**Pull request merged**